### ISSUE TITLE
Smooth-scrollbar not handle browser built-in scroll events

### ISSUE THREAD

**Issue by vitald88** [idiotWu/smooth-scrollbar#215]
Modern browser has include form verifying mechanism. In simple words you can mark input as required and browser will show message near it when it left empty when user try to submit form. Problem happens when form is longer than screen height and invalid input is not seen. Browser try to scroll page to show input but smooth-scrollbar not handle it. Also similar situation happens when I use built-in search (Ctrl+F or F3) in browser on page with smooth-scrollbar. Is it possible to handle this kind of system scroll events on smooth-scrollbar?

**Comment by xootix** [comment]
I too have the same issue. Doesn't scroll on HTML validation.
@idiotWu  Please look into this.

**Comment by idiotWu** [comment]
I'm afraid such native features would not be supported with this plugin (nor do other custom scrollbars) 😥

**Comment by xootix** [comment]
Too bad !
@vitald88 , to scroll to input which has failed validation in a form, you can follow my way.
You can also use "invalid" event for more flexibility.

`$('.form-submit-button').on( 'click', function(){
	//Find all inputs which have failed invalidation
	var invalid_inputs = $(this).closest('form').find('input:invalid');
	if( invalid_inputs.length === 0 ) return;
	//Get scrollbar and scroll to first input which has failed invalidation
	Scrollbar.get( $(this).closest('.scrollbar-container').get(0)).scrollTop = invalid_inputs.filter(":first").position().top;
} );`

**Comment by xootix** [comment]
Too bad !
@vitald88 , to scroll to input which has failed validation in a form, you can follow my way.
You can also use "invalid" event for more flexibility.

https://pastebin.com/kmRmLcBQ

**Comment by vitald88** [comment]
> Too bad !
> @vitald88 , to scroll to input which has failed validation in a form, you can follow my way.
> You can also use "invalid" event for more flexibility.
> 
> https://pastebin.com/QX6HtTg1

Yep, thanks. I use similar code to do same but in vanilla js. This situation is just one of possible problems, all of them depends on not handling native/built-in scroll events by custom scrollbar, but if @idiotWu says that it's technically impossible it mean that every situation needs some sort of workaround.
